### PR TITLE
refactor: make aad addon user-configurable

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This PR creates an `aad` user-configurable addon, exposed via the existing `kubernetesConfig.addons` interface:

```
...
    "addons": [
        {
            "name": "aad",
            "enabled": true
        }
    ]
...
```

Strictly speaking, the above doesn't make sense in the absence of an `aadProfile` w/ `adminGroupID` (validation has been added for that), but this change makes it more explicit that the aad ClusterRoleBinding is being delivered via cloud-init to `/etc/kubernetes/addons` for reconciliation by `kube-addon-manager`; in addition, putting this into the user-configurable addons space makes future evolution/maintenance of AAD in the k8s application layer more tractable.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: parts/k8s/kubernetesparams.t [hunk 1]
```diff
@@ -1,19 +1,3 @@
-{{if .HasAadProfile}}
-    "aadTenantId": {
```
Comment (jackfrancis) [inline parts/k8s/kubernetesparams.t]: The aadTenantId is not referenced anywhere (probably leftover from a prior implementation), so I've cleaned it up while I'm in this area.

Checked as well that the AKS codebase doesn't depend on this in an unexpected way.

File: pkg/api/vlabs/validate.go [hunk 2]
```diff
@@ -636,9 +636,10 @@ func (a *Properties) validateAddons() error {
 				}
 			}
 
-			switch addon.Name {
-			case "cluster-autoscaler":
-				if to.Bool(addon.Enabled) {
+			// Validation for addons if they are enabled
+			if to.Bool(addon.Enabled) {
```
Comment (jackfrancis) [inline pkg/api/vlabs/validate.go]: Wrapping this entire case/switch in a "is this addon enabled?" condition fixed a really nasty bug due to my adding aad addon validation here. Essentially, the validation below is (almost) entirely only valid in the context of the addon being enabled, so we also get to remove the `if to.Bool(addon.Enabled) {` for each addon case.

There was one "if addon is disabled" validation for `cloud-node-manager`, which was moved out to a new `else` block below.

File: pkg/api/types_test.go [hunk 3]
```diff
@@ -6063,6 +6063,57 @@ func TestGetKubeProxyFeatureGates(t *testing.T) {
...
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			hasAADAdminGroupID := test.properties.HasAADAdminGroupID()
+			if hasAADAdminGroupID != test.expectedHasAADAdminGroupID {
+				t.Errorf("expected featureGates %t, but got %t", test.expectedHasAADAdminGroupID, hasAADAdminGroupID)
```
Comment (mboersma) [inline pkg/api/types_test.go]: Should this error message say `"expected hasAADAdminGroupID %t"` instead?

File: pkg/api/types_test.go [hunk 4]
```diff
@@ -6063,6 +6063,57 @@ func TestGetKubeProxyFeatureGates(t *testing.T) {
...
+			}
+			getAADAdminGroupID := test.properties.GetAADAdminGroupID()
+			if getAADAdminGroupID != test.expectedGetAADAdminGroupID {
+				t.Errorf("expected featureGates %s, but got %s", test.expectedGetAADAdminGroupID, getAADAdminGroupID)
```
Comment (mboersma) [inline pkg/api/types_test.go]: Same comment as above.

File: pkg/api/vlabs/types_test.go [hunk 5]
```diff
@@ -729,3 +729,54 @@ func GetMockPropertiesWithCustomCloudProfile(name string, hasCustomCloudProfile,
...
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			hasAADAdminGroupID := test.properties.HasAADAdminGroupID()
+			if hasAADAdminGroupID != test.expectedHasAADAdminGroupID {
+				t.Errorf("expected featureGates %t, but got %t", test.expectedHasAADAdminGroupID, hasAADAdminGroupID)
```
Comment (mboersma) [inline pkg/api/vlabs/types_test.go]: Ditto.

File: pkg/api/vlabs/types_test.go [hunk 6]
```diff
@@ -729,3 +729,54 @@ func GetMockPropertiesWithCustomCloudProfile(name string, hasCustomCloudProfile,
...
+			}
+			getAADAdminGroupID := test.properties.GetAADAdminGroupID()
+			if getAADAdminGroupID != test.expectedGetAADAdminGroupID {
+				t.Errorf("expected featureGates %s, but got %s", test.expectedGetAADAdminGroupID, getAADAdminGroupID)
```
Comment (mboersma) [inline pkg/api/vlabs/types_test.go]: Ditto.

**Comment by jackfrancis** [comment]
Related to #2251

**Pull request merged**